mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -568,6 +575,15 @@ impl MmArc {
             .map_err(|e| MmMetricsError::Internal(e.to_string()))?;
         prometheus::spawn_prometheus_exporter(self.metrics.weak(), address, shutdown_detector, credentials)
     }
+
+    pub fn get_scripthash_notification_handlers(&self) -> ScripthashNotificationHandlers {
+        if self.event_stream_configuration.is_some() {
+            let (sender, receiver): (AsyncSender<String>, AsyncReceiver<String>) = futures::channel::mpsc::channel(1);
```

Comment:
Again, 1 might be not enough: depending on frequency of notifications (especially if we subscribe for multiple script hashes), the receiving part won't be able to process messages fast enough. I'm not sure about what value we should use here, but a number of subscribed addresses might be optimal here.

Reply:
Used unbounded channels as any static value can be insufficient or exceeding. Only concern is when mm2 connects to electrum client which is malicious and sends hundreds of millions of scripthash notifications per second. But at that point we have bigger problem than running out of memory.

Reply:
> Used unbounded channels as any static value can be insufficient or exceeding

I didn't propose using unbounded channels due to this. The idea was to have increased capacity to avoid losing notifications for different script hashes, if they occasionally arrive concurrently. Of course, if server sends them "too fast", any capacity will be exceeded. Though, there will be lower chance to lose a notification due to "channel is full". Anyway, it's quite theoretical, so maybe it's worth trying to perform a kind of stress test of this implementation. It might turn out that even with capacity set to 1 we won't have any issues in real use cases.

Reply:
Unbounded channels will handle this dynamically without needing to define channel capacity. The capacity need can be very different anytime on runtime depending on the user account and activity.